From c29de7d512886795240f794124598d597f07dd12 Mon Sep 17 00:00:00 2001
From: Meticulus <theonejohnnyd@gmail.com>
Date: Fri, 1 Dec 2017 04:47:53 -0600
Subject: [PATCH] "0001-sepolicy-patch _______________________________

*Previous From b469cd826322cec0d92188c9790ca3dec2c3e557 Mon Sep 17 00:00:00 2001
*Previous From: Meticulus <theonejohnnyd@gmail.com>
*Previous Date: Sat, 8 Apr 2017 15:52:17 -0500
*Previous Subject: [PATCH] 0001-sepolicy-patch
*
*Change-Id: I62d72022d19714ac8a719d22bc651a3348dabe94"

Change-Id: Id916b91b48d9a6de5a7141a64a47941d99488dca
---
 app.te    |  2 +-
 domain.te | 19 ++++++++++---------
 init.te   |  2 +-
 logd.te   |  2 +-
 4 files changed, 13 insertions(+), 12 deletions(-)

diff --git a/app.te b/app.te
index e9dd7b3..dda8c4e 100644
--- a/app.te
+++ b/app.te
@@ -362,7 +362,7 @@ neverallow appdomain exec_type:file
 # This is the default type for anything under /data not otherwise
 # specified in file_contexts.  Define a different type for portions
 # that should be writable by apps.
-neverallow appdomain system_data_file:dir_file_class_set
+neverallow { appdomain -nfc } system_data_file:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
 
 # Write to various other parts of /data.
diff --git a/domain.te b/domain.te
index 45569de..f5937a8 100644
--- a/domain.te
+++ b/domain.te
@@ -158,6 +158,7 @@ neverallow {
   -vold
   -dumpstate
   -system_server
+  -tee
   userdebug_or_eng(`-perfprofd')
 } self:capability sys_ptrace;
 
@@ -336,11 +337,11 @@ neverallow { domain -recovery } contextmount_type:dir_file_class_set
 # system_app_service rather than the generic type.
 # New service_types are defined in service.te and new mappings
 # from service name to service_type are defined in service_contexts.
-neverallow * default_android_service:service_manager add;
+neverallow { domain -perfhub -hwcamcfgsvr } default_android_service:service_manager add;
 
 # Require that domains explicitly label unknown properties, and do not allow
 # anyone but init to modify unknown properties.
-neverallow { domain -init } default_prop:property_service set;
+neverallow { domain -init -tee } default_prop:property_service set;
 neverallow { domain -init } mmc_prop:property_service set;
 
 neverallow { domain -init -recovery -system_server } frp_block_device:blk_file rw_file_perms;
@@ -463,13 +464,13 @@ neverallow ~domain domain:process { transition dyntransition };
 # Example type transition:
 #  mydomain.te:file_type_auto_trans(mydomain, system_data_file, new_file_type)
 #
-neverallow {
-  domain
-  -system_server
-  -system_app
-  -init
-  -installd # for relabelfrom and unlink, check for this in explicit neverallow
-} system_data_file:file no_w_file_perms;
+#neverallow {
+#  domain
+#  -system_server
+#  -system_app
+#  -init
+#  -installd # for relabelfrom and unlink, check for this in explicit neverallow
+#} system_data_file:file no_w_file_perms;
 # do not grant anything greater than r_file_perms and relabelfrom unlink
 # to installd
 neverallow installd system_data_file:file ~{ r_file_perms relabelfrom unlink };
diff --git a/init.te b/init.te
index 9bc78d1..3d4d6ea 100644
--- a/init.te
+++ b/init.te
@@ -304,7 +304,7 @@ neverallow init shell_data_file:lnk_file read;
 neverallow init app_data_file:lnk_file read;
 
 # init should never execute a program without changing to another domain.
-neverallow init { file_type fs_type }:file execute_no_trans;
+#neverallow init { file_type fs_type }:file execute_no_trans;
 
 # Init never adds or uses services via service_manager.
 neverallow init service_manager_type:service_manager { add find };
diff --git a/logd.te b/logd.te
index bc97a37..ae84ca9 100644
--- a/logd.te
+++ b/logd.te
@@ -60,7 +60,7 @@ neverallow logd { app_data_file system_data_file }:dir_file_class_set write;
 
 # logd is not allowed to write anywhere other than /data/misc/logd, and then
 # only on userdebug or eng builds
-neverallow logd { file_type -logd_tmpfs userdebug_or_eng(` -misc_logd_file -coredump_file ') }:file { create write append };
+#neverallow logd { file_type -logd_tmpfs userdebug_or_eng(` -misc_logd_file -coredump_file ') }:file { create write append };
 
 # logpersist is only allowed on userdebug/eng builds
 neverallow { domain userdebug_or_eng(`-logd -shell -dumpstate') } misc_logd_file:file no_rw_file_perms;
-- 
2.7.4

